src/core/http-request/http-request-impl.ts

```diff
@@ -29,7 +33,7 @@ export class HttpRequestImpl implements HttpRequest {
       ...this.headers,
       [header]: value
     };
-    return new HttpRequestImpl(this.url, this.body, this.method, newHeaders);
+    return new HttpRequestImpl(this.url, this.method, this.body, newHeaders);
```

Comment:
What are the criteria for ordering the parameters in this way?

Reply:
`url` and `method` are mandatory but a request without body or any header is still a valid request.